component/src/main/java/org/wso2/extension/siddhi/io/jms/sink/JMSSink.java

```diff
@@ -132,35 +133,25 @@ protected void init(StreamDefinition outputStreamDefinition, OptionHolder option
 
     @Override
     public void connect() throws ConnectionUnavailableException {
-        this.clientConnector = new JMSClientConnector();
-    }
-
-    @Override
-    public Map<String, Object> currentState() {
-        return null;
-    }
-
-    @Override
-    public void restoreState(Map<String, Object> state) {
-
+        try {
+            this.clientConnector = new JMSConnectorFactoryImpl().createClientConnector(jmsStaticProperties);
+        } catch (JMSConnectorException e) {
+            log.error("Error connecting to JMS provider. " + e.getMessage());
```

Comment:
Should we not throw ConnectionUnavailableException here?

Reply:
Fixed with commit [#37ec923ea45f85854937623969a5960c1c5cc5a7](https://github.com/wso2-extensions/siddhi-io-jms/pull/25/commits/37ec923ea45f85854937623969a5960c1c5cc5a7)